internal/controllers/synthesis/lifecycle.go

```diff
@@ -318,10 +318,10 @@ func shouldDeletePod(logger logr.Logger, comp *apiv1.Composition, syn *apiv1.Syn
 	return logger, nil, false
 }
 
-func swapStates(comp *apiv1.Composition) {
-	// If the previous state has been synthesized but not the current, keep the previous to avoid orphaning deleted resources
-	if comp.Status.CurrentSynthesis != nil && comp.Status.CurrentSynthesis.Synthesized != nil {
-		comp.Status.PreviousSynthesis = comp.Status.CurrentSynthesis
+func SwapStates(comp *apiv1.Composition) {
+	current := comp.Status.CurrentSynthesis
+	if current != nil && current.Synthesized != nil && !current.Failed() {
```

Comment:
This prevents us from losing the last known good state - otherwise we may end up orphaning resources, changing properties we shouldn't (due to missing three way merge metadata), etc.

Reply:
Essentially this is causing the failed synthesis to "stay current" and be overwritten with an "empty" synthesis?

Reply:
Yep exactly. Otherwise we might push out a non-empty synthesis and lose the three-way merge state it contains.